### ISSUE TITLE
Update improper use of `yield` in data stores

### ISSUE THREAD

**Issue by nfmohit** [google/site-kit-wp#6117]
## Feature Description

There are a few examples of improper (although non-breaking) generators in our data stores that should be cleaned up.

<details>
<summary>Example / original description</summary>

https://github.com/google/site-kit-wp/blob/5a22b076d4bfe47f44f0a8455253a19b056e4fef/assets/js/googlesitekit/modules/datastore/modules.js#L490

This `yield` is unnecessary because it's dispatching directly. If the intention is to await the action, we'd need to use `yield Data.commonActions.await()` around it since the dispatched action will return a promise. In this case, I think we can simply remove it as we don't depend on the response?

Ideally, it might be good to use the await action with a `Promise.all` so these can be fetched in parallel.

_Originally posted by @aaemnnosttv in https://github.com/google/site-kit-wp/pull/5988#discussion_r1014495179_

</details>

      

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* `yield` use within data store generator functions should be updated such that only "raw" action objects are yielded from the generator
	* More specifically, the result of a call to the `registry`'s `dispatch` should not be yielded as these always return a `Promise`

## Implementation Brief

* Find all usage of `yield (registry\.)?dispatch` calls
	* When dispatching an action **on the same store**, replace the `dispatch` call to the unbound action creator (i.e. `actions.someAction()`)
	* For instances where the `storeName` is dynamic or on another store:
		* If the dispatched action is async (involves a control), keep the `yield` but wrap the call to `dispatch` in the common `await` action
		* Otherwise, if it is a simple non-async action (e.g. `invalidateResolution`), keep the `dispatch` call and remove the `yield`

### Test Coverage

* No changes expected

## QA Brief

- This doesn't involve any feature/interface changes.
- Perform smoke testing and ensure no regressions due to these changes.
- Ensure Recoverable modules and User Survey features work as expected.

## Changelog entry

* Improve async handling of some data store actions.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- Performed smoke testing and found no regressions due to these changes.
  - Setup Site Kit with Analytics.  Then, enabled and disabled all modules. 
 -  Recoverable modules and User Survey features work as expected.

**Comment by aaemnnosttv** [comment]
## ⚠️ Approval 

I found an instance that was missed
https://github.com/google/site-kit-wp/blob/cf7d5bc66b934502eb2494cba7621f763d312550/assets/js/googlesitekit/modules/datastore/modules.js#L391-L396

This should be very quick to clean up, so let's address this in a follow-up PR.

**Comment by aaemnnosttv** [comment]
## ✅ Approval

Thanks @tofumatt 👍
